### ISSUE TITLE
Completely Disabling Antiforgery

### ISSUE THREAD

**Issue by Jon-Murray** [aspnet/Mvc#7396]
Hi,

   I'm trying to completely disable antiforgery, but it seems this is not possible. The documentation seems to suggest that using AddMvcCore as the only service will disable anti forgery. Whilst this seems true in that the tokens are not validated, the .aspnet.antiforgery token still gets set in any requests omitting it. 

Trying to manually remove any services which are anti forgery related will throw an exception when attempting to view a view, such as:

InvalidOperationException: Unable to resolve service for type 'Microsoft.AspNetCore.Antiforgery.IAntiforgery' while attempting to activate 'Microsoft.AspNetCore.Mvc.ViewFeatures.DefaultHtmlGenerator'.

Is there any way to completely get rid of the cookies issued? I know i can rename them, but this isn't really the point.

Thanks!

**Comment by Lechus** [comment]
Web API does not need to set/send it client, does it?

**Comment by mkArtakMSFT** [comment]
@javiercn, any suggestions here?

**Comment by javiercn** [comment]
@Jon-Murray Can you provide more details on what you are doing on your startup class?
```csharp
services.AddMvcCore()...
```
Seems like you are using Views in your app. Calling  
`services.AddMvcCore().AddViews()` will bring antiforgery services along and some of our components in our views (form taghelper) generate antiforgery tokens automatically for you, which is the likely reason you are seeing the cookie being set.

**I EXTREMELY discourage you from disabling antiforgery** in your app if you are developing any sort of app that is accessible through the browser or accepts form encoded content, but if you choose to do so I do believe you can disable antiforgery generation on a per form basis by adding asp-antiforgery="false" to the form or globally through a taghelper initializer.

With that, you shouldn't see any cookie.

So to recap:
* If you are not using views, you should not see a cookie.
* If you disable antiforgery on the form tag helper and (potentially) other components that generate it automatically, you shouldn't see the cookie.

**Comment by winzig** [comment]
> or globally through a taghelper initializer

Can anyone explain how to disable it globally as mentioned above?
